src/classes/CRLP_Batch_Base.cls

```diff
@@ -350,19 +353,22 @@ public abstract class CRLP_Batch_Base {
     /******************************** Helper Methods *****************************/
 
     /**
-     * @description Builds and returns batch job start query string.
-     * @param chunkLastId The last processed Id from the previous chunk.
+     * @description Builds and returns batch job start query string.  The BatchQueryBuilder instance
+     * needs to be constructed here in the top-level CRLP_Batch_Base class so that a reference to it
+     * can be passed in to the BatchQueryBuilder constructor, making the top-level class variables
+     * accessible to the inner class. Used in the start method of the CRLP_Batch_Base_NonSkew class.
+     * @param lastIdProcessed The last processed Id from the previous chunk.
      * @param chunkSize The size of each chunk - used in the LIMIT statement in the query.
      * @param fieldsToCheckForNonZero A list of fields that were checked for non-zero values,
      * used in the query.
      * @return The query string to be used by the batch start method.
      */
     @TestVisible
-    protected String getBatchStartQuery(Id chunkLastId, Integer chunkSize, List<String> fieldsToCheckForNonZero) {
+    protected String getBatchStartQuery(Id lastIdProcessed, Integer chunkSize, List<String> fieldsToCheckForNonZero) {
```

Comment:
@kslewis  Since `fieldsToCheckForNonZero` is a specific name that describes the usage (which is not necessarily known in this wrapper class), how about changing the var name here (and in the inner class) to `addlFieldsToQuery`.

Reply:
Also, does this need to pass in `chunkSize` as a parameter or is thing something the wrapper class could determine on its own using the current job type?

Reply:
Yes, it wouldn't need to be passed in, but I'm also using the constructor param to pass in a limit for the query that checks for additional records in `hasAdditionalRecordsToProcess`. 

I renamed the fieldsToCheckForNonZero variable!